zerver/views/streams.py

```diff
@@ -410,6 +409,9 @@ def stream_exists_backend(request, user_profile, stream_name, autosubscribe):
     stream = get_stream(stream_name, user_profile.realm)
     result = {"exists": bool(stream)}
     if stream is not None:
+        if stream.invite_only and stream.num_subscribers() == 0:
+            do_deactivate_stream(stream)
+            return json_response(data=result, status=404)
```

Comment:
@timabbott ,

If deactivate the invite-olny stream when the laster user leaves, it is unavailable to create a new stream with deactivated stream name.